routes/clients.rb

```diff
@@ -18,11 +18,33 @@ def self.registered(app)
       app.get '/clients/:id/?' do
         # view a client
         @client = Client[id: params[:id]]
+        @project = Project[id: params[:id]]
+        @node = NodeResource[id: params[:id]]
+        @db = DBResource[id: params[:id]]
+
         if @client.nil?
           MyLog.log.fatal 'routes/clients: Client not found'
           halt 404, 'Client not found'
         end
-        @projects = Project.filter(client_id: @client.id)
+
+        if @project.nil?
```

Comment:
I don't think you need this if/else - just fetch the projects from this client, (client.projects)  and if it's an empty dataset, you can use that as a condition for the if in the template (<% if @projects %>Stuff...)

similarly with node_resources and db_resources